File: deepcoffee-api/app/schemas/equipment.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


EquipmentCategory = Literal["brewer", "grinder", "filter_media", "water"]


class EquipmentItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    category: EquipmentCategory
    name: str
    notes: str | None = None
    is_default: bool = False
    created_at: datetime
    updated_at: datetime


class EquipmentCreateRequest(BaseModel):
    category: EquipmentCategory
    name: str = Field(min_length=1, max_length=120)
    notes: str | None = Field(default=None, max_length=500)
    is_default: bool | None = None


class EquipmentUpdateRequest(BaseModel):
    category: EquipmentCategory | None = None
    name: str | None = Field(default=None, min_length=1, max_length=120)
    notes: str | None = Field(default=None, max_length=500)
    is_default: bool | None = None

File: frontend/lib/api/equipment.ts
// 我的器具 API（/v1/equipment）。当前模型是按类别保存的单件器具库存。
import { apiFetch } from './client'

export type EquipmentCategory = 'brewer' | 'grinder' | 'filter_media' | 'water'

export interface EquipmentProfile {
  id: string
  category: EquipmentCategory
  name: string
  notes?: string | null
  is_default: boolean
  created_at: string
  updated_at: string
}

export interface EquipmentInput {
  category?: EquipmentCategory
  name?: string
  notes?: string
  is_default?: boolean
}

export function listEquipment(): Promise<EquipmentProfile[]> {
  return apiFetch('/equipment')
}

export function createEquipment(body: EquipmentInput): Promise<EquipmentProfile> {
  return apiFetch('/equipment', { method: 'POST', body: JSON.stringify(body) })
}

export function updateEquipment(id: string, body: EquipmentInput): Promise<EquipmentProfile> {
  return apiFetch(`/equipment/${encodeURIComponent(id)}`, { method: 'PATCH', body: JSON.stringify(body) })
}

export function deleteEquipment(id: string): Promise<{ deleted: boolean }> {
  return apiFetch(`/equipment/${encodeURIComponent(id)}`, { method: 'DELETE' })
}

// 设为该类别默认项；后端自动取消同类别其余默认。
export function setDefaultEquipment(id: string): Promise<EquipmentProfile> {
  return updateEquipment(id, { is_default: true })
}
